### ISSUE TITLE
About the Freihand demo

### ISSUE THREAD

**Issue by hungsing92** [mks0601/I2L-MeshNet_RELEASE#23]
Thanks for your great work!
I want to modify your demo.py to check FreiHand model. But I found it needs the camera focal parameter and the root_depth. I guess them from the freihand annotations, but I think that is not accurate. Can you give me some help? And I may check your model performance on other hands benchmark.

**Comment by mks0601** [comment]
I used camera parameters from FreiHAND annotations, which is available in the evaluation setting.
The root depth is obtained from RootNet, my ICCV19 paper [link](https://github.com/mks0601/3DMPPE_ROOTNET_RELEASE). You can download the depth values from [here](https://drive.google.com/drive/folders/1OzwQG2ZutJ4Lzg5psilStsv_MO6-cysA?usp=sharing).

**Comment by hungsing92** [comment]
What if I want to run on other images not from FreiHand? And can you share the pretrained model of RootNet?

**Comment by mks0601** [comment]
Sorry I didn't save the pre-trained RootNet on FreiHAND.
If you do not know the intrinsics, you can just use normalized intrinsics (fx=1500, fy=1500, cx=img_width/2, cy = img_height/2).
Or, if you assume a weak-perspective, you don't need the focal lengths because they are eliminated during camera-backprojeciton.

**Comment by hungsing92** [comment]
I got  a bad result by using  (fx=1500, fy=1500, cx=img_width/2, cy = img_height/2).  (fx=500, fy=500, cx=img_width/2, cy = img_height/2) is much better.

**Comment by mks0601** [comment]
good. you can use a weak-perspective camera model to eliminate the focal lengths.

**Comment by hungsing92** [comment]
@mks0601 I test more hand pictures, but they seems not good. Can you give some suggestions?

**Comment by mks0601** [comment]
You'd better visualize `mesh_out_img` at [here](https://github.com/mks0601/I2L-MeshNet_RELEASE/blob/21a8930d8c0a8815565940f9c5714d43cc0fe9c8/data/FreiHAND/FreiHAND.py#L188). This 2.5D representation does not require intrinsics and root depth

**Comment by hungsing92** [comment]
@mks0601 I used function "vis_mesh" to visualize the x,y. But It's still not good.

**Comment by mks0601** [comment]
I can't give you more detailed guidance because I don't know your testing images

**Comment by vodanhbk95** [comment]
@mks0601 
I test on FreiHand dataset with image like this 
![f659](https://user-images.githubusercontent.com/22610128/111939065-e3509480-8afd-11eb-844f-8dc1be3358ec.jpg)
And got this result 
![rendered_mesh_param_new](https://user-images.githubusercontent.com/22610128/111939119-fd8a7280-8afd-11eb-8814-d11e192ef9a3.jpg)
How can I improve result on Hand dataset??

**Comment by mks0601** [comment]
You used a body network.. please read the README carefully and use a hand network.

**Comment by vodanhbk95** [comment]
@mks0601  Sorry but your mean is using checkpoint snapshot_24.pth.tar (of FreiHAND), right?
Because when I used it and ran demo with command <python demo.py --gpu 1 --stage param --test_epoch 24>, it got error size mismatch for module.human_model_layer.th_faces, module.human_model_layer.th_weights

**Comment by mks0601** [comment]
You'd better just run a full code, not a demo code if you want to run the hand model. There should be several modifications for the hand demo code. If you want to run the demo code, could you set 
```
trainset_3d = ['FreiHAND'] # MuCo, Human36M, FreiHAND
trainset_2d = [] # MSCOCO
```

in the `main/config.py`?

**Comment by vodanhbk95** [comment]
@mks0601 Surely, I have modified this config file. test_set I set for FreiHAND too :( . I am so confused about this problem

**Comment by vodanhbk95** [comment]
@mks0601 . After investigating many times. I get this results. Did I do right??
![output_mesh_lixel](https://user-images.githubusercontent.com/22610128/112100152-3fd3b280-8bd7-11eb-87f4-92ddbc6d2edb.jpg)
![rendered_mesh_lixel](https://user-images.githubusercontent.com/22610128/112100171-46622a00-8bd7-11eb-93c6-a2ad9fbcec22.jpg)

**Comment by mks0601** [comment]
seems like you did right!
